### ISSUE TITLE
Segfault on display re-init

### ISSUE THREAD

**Issue by robertpfeiffer** [pygame/pygame#2343]
Ubuntu 20.04, SDL2.0.12 compiled from source

minimal example:
```python
import pygame
pygame.init()
pygame.display.init()
pygame.display.quit()
pygame.display.init()
```

backtrace:
```
(gdb) bt
#0  0x0000000000000000 in  ()
#1  0x00007ffff790d54c in SDL_DBus_QueryPropertyOnConnection
    (conn=<optimized out>, node=node@entry=0x7ffff79415d1 "org.freedesktop.RealtimeKit1", path=path@entry=0x7ffff79415ee "/org/freedesktop/RealtimeKit1", interface=<optimized out>, interface@entry=0x7ffff79415d1 "org.freedesktop.RealtimeKit1", property=<optimized out>, 
    property@entry=0x7ffff79415c4 "MinNiceLevel", expectedtype=expectedtype@entry=105, result=0x7ffff7978640 <rtkit_min_nice_level>)
    at /home/rincewind/gamedev/SDL2-2.0.12/src/core/linux/SDL_dbus.c:304
#2  0x00007ffff7917f75 in rtkit_initialize () at /home/rincewind/gamedev/SDL2-2.0.12/src/core/linux/SDL_threadprio.c:51
#3  0x00007ffff7d8e47f in __pthread_once_slow (once_control=0x7ffff797e328 <rtkit_initialize_once>, init_routine=0x7ffff7917f30 <rtkit_initialize>)
    at pthread_once.c:116
#4  0x00007ffff7d8e535 in __GI___pthread_once
    (once_control=once_control@entry=0x7ffff797e328 <rtkit_initialize_once>, init_routine=init_routine@entry=0x7ffff7917f30 <rtkit_initialize>)
    at pthread_once.c:143
#5  0x00007ffff7918007 in rtkit_setpriority (nice_level=-20, thread=<optimized out>)
    at /home/rincewind/gamedev/SDL2-2.0.12/src/core/linux/SDL_threadprio.c:64
#6  SDL_LinuxSetThreadPriority_REAL (threadID=<optimized out>, priority=-20)
    at /home/rincewind/gamedev/SDL2-2.0.12/src/core/linux/SDL_threadprio.c:105
#7  0x00007ffff790ec6f in SDL_SYS_SetThreadPriority (priority=priority@entry=SDL_THREAD_PRIORITY_TIME_CRITICAL)
    at /home/rincewind/gamedev/SDL2-2.0.12/src/thread/pthread/SDL_systhread.c:207
#8  0x00007ffff787ad79 in SDL_SetThreadPriority_REAL (priority=priority@entry=SDL_THREAD_PRIORITY_TIME_CRITICAL)
    at /home/rincewind/gamedev/SDL2-2.0.12/src/thread/SDL_thread.c:470
#9  0x00007ffff7805be4 in SDL_RunAudio (devicep=devicep@entry=0xcd7c40) at /home/rincewind/gamedev/SDL2-2.0.12/src/audio/SDL_audio.c:700
#10 0x00007ffff787a9f0 in SDL_RunThread (data=0xcd90a0) at /home/rincewind/gamedev/SDL2-2.0.12/src/thread/SDL_thread.c:289
#11 0x00007ffff790ea6d in RunThread (data=<optimized out>) at /home/rincewind/gamedev/SDL2-2.0.12/src/thread/pthread/SDL_systhread.c:79
#12 0x00007ffff7d85609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#13 0x00007ffff7ec1293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
```

**Comment by MyreMylar** [comment]
Just checked and this doesn't segfault on windows with python 3.9 64bit and 2.0.1.dev1. 

I'd be interested to see if it is ubuntu 64bit SDL & python only or whether it happens on all flavours of linux and 32bit.

**Comment by ankith26** [comment]
Can reproduce segfault, raspberry pi OS on pi3
```
pygame 2.0.1.dev1 (SDL 2.0.9, python 3.7.3)
```
(side note, pygame built from my PR branch, but I didnt mess with any display stuff)

**Comment by robertpfeiffer** [comment]
This is related to event posting @ankith26. If we figure this one out, we can probably fix the event filter once and for all, but as of now, every way of setting the event filter I tried interacted weirdly with init, which is why I took that code out of the other PR.

Also, I should probably post a traceback of the other threads...

**Comment by ankith26** [comment]
I don’t know how this display stuff is related to event stuff. Will fixing this bug also fix the event-filter setting issue?

PS: I’ve come up with a few good solutions to unicode issues and event posting issues. It’s ready locally. Just doing a bit of testing.

**Comment by ankith26** [comment]
Another interesting observation
```python
import pygame
pygame.init()
pygame.display.quit()
print("This line surprisingly gets printed")
```
The segfault still happens, but the print statement still gets executed.

**Comment by bydariogamer** [comment]
```~$ python
Python 3.10.4 (main, Mar 23 2022, 23:05:40) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pygame
pygame 2.1.2 (SDL 2.0.16, Python 3.10.4)
Hello from the pygame community. https://www.pygame.org/contribute.html
>>> pygame.init()
(5, 0)
>>> pygame.quit()
>>> pygame.init()
(5, 0)
>>> pygame.quit()
>>> pygame.init()
(5, 0)
>>> pygame.quit()
>>> pygame.init()
(5, 0)
>>> pygame.init()
(5, 0)
>>> pygame.quit()
>>> pygame.quit()
>>> pygame.init()
(5, 0)
```
No segfaults anymore. I think this issue was solved with the numerous changes that affected the initialization.

**Comment by ankith26** [comment]
Re-visiting this issue after a while, I can confirm bydariogamer's report of this issue being fixed on 2.1.3dev5, on my pi3 (which I had reproduced this bug on about 2 years ago now). But the platform is not exactly the same: I changed to a 64-bit PiOS a few months ago (which means I can get to using the manylinux aarch64 wheels that pygame provides). I suspect this was either fixed with the large bump in SDL version from then to now, or the SDL_mixer bump, or both.

I'll go ahead and close this, please feel free to re-open if this bug is still reproducible in some platform configuration.
